doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst

```diff
@@ -0,0 +1,322 @@
+#. Other devices will then join an existing network during this join period.
+
+    * When a device has joined and ``zarco_network_rejoin`` is running, the procedure is cancelled.
+    * If no device has joined and the procedure is not running, the procedure will be started.
+
+   .. figure:: /images/zigbee_signal_handler_06_steering.png
+      :alt: Forming a network following the generation of ZB_BDB_SIGNAL_STEERING
+
+      Forming a network following the generation of ZB_BDB_SIGNAL_STEERING
+
+.. _zarco_signal_handler_leave:
+
+Zigbee network leaving
+======================
+
+The default signal handler implements the same behavior for handling ``ZB_ZDO_SIGNAL_LEAVE`` for both routers and end devices.
+When leaving the network, the default handler calls :cpp:func:`start_network_rejoin()` to start ``zarco_network_rejoin`` to join a new network.
+
+Once :cpp:func:`start_network_rejoin()` is called, the stack will generate the ``ZB_BDB_SIGNAL_STEERING`` signal and will continue to ``zarco_signal_handler_network``.
+
+.. figure:: /images/zigbee_signal_handler_09_leave.png
+   :alt: Leaving the network following ZB_ZDO_SIGNAL_LEAVE
+
+   Leaving the network following ZB_ZDO_SIGNAL_LEAVE
+
+.. _zarco_network_rejoin:
+
+Zigbee network rejoining
+========================
+
+The Zigee network rejoin procedure is a mechanism that is similar to the ZDO rejoin back-off procedure.
+It is implemented to work with both routers and end devices and simplify handling of cases such as device joining, rejoining, or leaving the network.
+It is used in :cpp:func:`default_signal_handler()` by default.
+
+If the network is left by a router or an end device, the device will try to join any open network.
+
+* The router will use the default signal handler to try to join or rejoin the network until it succeeds.
+* The end device will use the default signal handler to try to join or rejoin the network for a finite period of time, because the end devices are often powered by batteries.
+
+  * The procedure to join or rejoin the network is restarted after the device reset or power cycle.
+  * The procedure to join or rejoin the network can be restarted by calling :cpp:func:`user_input_indicate()`, but it needs to be implemented in the application (for example, by calling :cpp:func:`user_input_indicate()` when a button is pressed).
+    The procedure will be restarted only if the device does not join and the procedure is not running.
+
+The Zigbee rejoin procedure retries to join a network with each try after a specified amount of time: ``2^n`` seconds, where ``n`` is the number of retries.
+
+The period is limited to 15 minutes if the result is higher than that.
+
+* When :cpp:func:`start_network_rejoin()` is called, the rejoin procedure is started, and depending on the device role:
+
+  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
```

Comment:
@sebastiandraus does ```ZB_DEV_REJOIN_TIMEOUT_MS``` is from KConfig options. If, so we should use it here.

Reply:
@sebastiandraus , any updates about this?

Reply:
No, the `ZB_DEV_REJOIN_TIMEOUT_MS` does not come from the KConfig.